builtin/logical/pki/path_fetch.go

```diff
@@ -179,6 +181,19 @@ func (b *backend) pathFetchRead(ctx context.Context, req *logical.Request, data
 			contentType = "application/pkix-cert"
 		}
 	case req.Path == "crl" || req.Path == "crl/pem":
+		if hasHeader(headerIfModifiedSince, req) {
+            // Maybe a sc declared at the top of function?
```

Comment:
Yeah I like hoisting this up and moving it earlier since the other branch also uses this, yeah.

Reply:
> This looks great! I think you'd want to do it for `pathGetIssuerCRL` and perhaps `pathGetRawIssuer` as well? :-) You might need to store the creation time on the `IssuerEntry` object as well, perhaps migrating if one doesn't exist. I wouldn't do the CA if-modified-since on the default issuer (`path_fetch.go`).

Hey, looked at both of the functions/endpoints? that you mentioned but I don't really understand why we would want to also add the IfModifiedSince check in `pathGetRawIssuer` as it does not seem to cover anything related to CRL.

Reply:
Ah, I can explain that better, but I'll leave it up to you if you want to do it. I'm happy with the scope of this PR if you don't. :-) 

So, the original request (#3190 linked above) said they had two use cases:

1. Auto-deploying the CRL.
2. Auto-deploying the CA. 

They want to avoid the "fetch and compare" with the existing approach (without these headers) and have Vault indicate if something has changed.

While you have addressed the first one (thanks!) the second one is a bit more subtle: we need to know not only the generation date of the CA (which isn't the same as the NotBefore/NotAfter on the cert) -- but also the last time the default was modified.

I was thinking that (in my original comment), as a substitute/proxy for the later (fetching `/cert/ca`), we could add this Header logic to `/issuer/:issuer_ref/{json,der,pem}` -- there, we'd need some field as a modification date. My suggestion was to add the generation date on the IssuerEntry to compare against...

However, this doesn't really make sense -- when fetching by issuer ID, the value of the CA there will never change. But when fetching by name, the generation date isn't the correct field, but I think instead a modification timestamp is the correct field.

If we had that field, we could update the IssuerEntry's LastModifiedTimestamp on both name change and on default ref change, and that'd allow the global `/cert/ca` or `/issuer/default` to also be fetched with the header.

Then, you can use the IfModifiedSince header to detect if the issuer has been rotated somehow, and only rarely have the same cert fetched (if you changed its name but its still default for instance). 

Let me know if that makes more sense? :-)

Reply:
Hello, I am aware of the original request but I thought, given your initial message, it was not necessary to do it for the CA endpoint(s). I don't mind including it as part of this PR just thought it was not the focus.
Regarding your explanation, it made sense and helped me create an initial version of the "check if modified" for the CA endpoints.
I still intend to take another look to see if there is a better way to do it, but feel free to take a look at it and drop some feedback. Thanks

Reply:
Sorry Gabriel! I was out on PTO again when you replied. I'll take a look today. :-)